vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/FileManagerTest.java

```diff
@@ -0,0 +1,143 @@
+package edu.asu.diging.vspace.core.services.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import edu.asu.diging.vspace.core.data.FileRepository;
+import edu.asu.diging.vspace.core.exception.FileStorageException;
+import edu.asu.diging.vspace.core.factory.IFileFactory;
+import edu.asu.diging.vspace.core.file.IStorageEngine;
+import edu.asu.diging.vspace.core.model.IVSFile;
+import edu.asu.diging.vspace.core.model.impl.VSFile;
+
+public class FileManagerTest {
+    
+    @Mock
+    private IStorageEngine storageEngine;
+    
+    @Mock
+    private IFileFactory fileFactory;
+    
+    @Mock
+    private FileRepository fileRepo;
+    
+    @InjectMocks
+    private FileManager serviceToTest;
+    
+    private final String fileId = "FILE_ID";
+    private final String fileName = "fileName";
+    private final String fileDescription = "fileName";
+    private final String fileContentString = "file content";
+    
+    
+    @Before
+    public void init() {
+        MockitoAnnotations.initMocks(this);
+    }
+    
+    @Test
+    public void test_getFileById_success() {
+        VSFile file = new VSFile();
+        when(fileRepo.findById(fileId)).thenReturn(Optional.of(file));
+        serviceToTest.getFileById(fileId);
+        verify(fileRepo).findById(fileId);
+    }
+    
+    @Test
+    public void test_getAllFiles_success() {
+        serviceToTest.getAllFiles();
+        verify(fileRepo).findAll();
+    }
+    
+    @Test
+    public void test_editFile_success() {
+        VSFile file = new VSFile();
+        file.setFilename(fileName);
+        when(fileRepo.findById(fileId)).thenReturn(Optional.of(file));
+        when(storageEngine.renameFile(file.getFilename(), fileName, null)).thenReturn(true);
+        IVSFile returnedFile = serviceToTest.editFile(fileId, fileName, fileDescription);
+        assertEquals(file.getFilename(), returnedFile.getFilename());
```

Comment:
what's being edited here? you need to check that the edited values are correct.

Reply:
I think this creates issues. The storage engine returns `file`. Presumably, `editFile` returns that same file object. So `file` == `returnedFile`, which means this line would test if `file.getFilename() is the same as file.isFilename()`. I would test it against the actual string (or test if `NEW_FILENAME` is not the same as `returnedFile.getFilename()`. Does that make sense?